### ISSUE TITLE
DragonProxy don't work when I chat in Minecraft PE

### ISSUE THREAD

**Issue by Hagb** [DragonetMC/DragonProxy#245]
I used the latest snapshots ( dadcc8adfacd7c7f3b7218018d380d9cdf2aa3f2 ) , Minecraft PE 1.12.10 and a Spigot 1.12.2 server.
When I chatted after I linked to DragonProxy , DragonProxy showed:
```
can not decode for pid 0x66
decode fail
java.lang.ArrayIndexOutOfBoundsException: 36
        at org.dragonet.common.utilities.BinaryStream.getByte(BinaryStream.java:197)
        at org.dragonet.common.utilities.VarInt.read(VarInt.java:186)
        at org.dragonet.common.utilities.VarInt.readUnsignedVarInt(VarInt.java:75)
        at org.dragonet.common.utilities.BinaryStream.getUnsignedVarInt(BinaryStream.java:380)
        at org.dragonet.common.utilities.BinaryStream.getByteArray(BinaryStream.java:362)
        at org.dragonet.common.utilities.BinaryStream.getString(BinaryStream.java:371)
        at org.dragonet.protocol.packets.TextPacket.decodePayload(TextPacket.java:69)
        at org.dragonet.protocol.PEPacket.decode(PEPacket.java:34)
        at org.dragonet.protocol.Protocol.decodeSingle(Protocol.java:138)
        at org.dragonet.protocol.Protocol.decode(Protocol.java:114)
        at org.dragonet.proxy.network.PEPacketProcessor.onTick(PEPacketProcessor.java:95)
        at org.dragonet.proxy.network.UpstreamSession.onTick(UpstreamSession.java:486)
        at org.dragonet.proxy.network.SessionRegister.onTick(SessionRegister.java:38)
        at org.dragonet.proxy.DragonProxy.onTick(DragonProxy.java:235)
        at org.dragonet.proxy.TickerThread.run(TickerThread.java:32)

```
and then nothing happened in my Spigot server and DragonProxy didn't work until I reconnected .
By the way, commands could be used without any exception , and only chatting caused exception.
Sorry for my bad English . ^_^

**Comment by HoverEpic** [comment]
Thank you for report, I think I badly update protocol, I'll take a look.

**Comment by patricknasci** [comment]
`java.io.EOFException
	at java.io.DataInputStream.readByte(DataInputStream.java:267)
	at org.dragonet.common.data.nbt.stream.NBTInputStream.readByte(NBTInputStream.java:65)
	at org.dragonet.common.data.nbt.tag.Tag.readNamedTag(Tag.java:85)
	at org.dragonet.common.data.nbt.tag.CompoundTag.load(CompoundTag.java:36)
	at org.dragonet.common.data.nbt.tag.ListTag.load(ListTag.java:50)
	at org.dragonet.common.data.nbt.tag.Tag.readNamedTag(Tag.java:94)
	at org.dragonet.common.data.nbt.tag.CompoundTag.load(CompoundTag.java:36)
	at org.dragonet.common.data.nbt.tag.Tag.readNamedTag(Tag.java:94)
	at org.dragonet.common.data.nbt.NBTIO.read(NBTIO.java:44)
	at org.dragonet.common.data.nbt.NBTIO.read(NBTIO.java:61)
	at org.dragonet.common.utilities.BinaryStream.getSlot(BinaryStream.java:310)
	at org.dragonet.protocol.packets.MobEquipmentPacket.decodePayload(MobEquipmentPacket.java:35)
	at org.dragonet.protocol.PEPacket.decode(PEPacket.java:34)
	at org.dragonet.protocol.Protocol.decodeSingle(Protocol.java:138)
	at org.dragonet.protocol.Protocol.decode(Protocol.java:114)
	at org.dragonet.proxy.network.PEPacketProcessor.onTick(PEPacketProcessor.java:95)
	at org.dragonet.proxy.network.UpstreamSession.onTick(UpstreamSession.java:486)
	at org.dragonet.proxy.network.SessionRegister.onTick(SessionRegister.java:38)
	at org.dragonet.proxy.DragonProxy.onTick(DragonProxy.java:235)
	at org.dragonet.proxy.TickerThread.run(TickerThread.java:32)
java.lang.ArrayIndexOutOfBoundsException: 303
	at org.dragonet.common.utilities.BinaryStream.getByte(BinaryStream.java:197)
	at org.dragonet.common.utilities.VarInt.read(VarInt.java:186)
	at org.dragonet.common.utilities.VarInt.readUnsignedVarInt(VarInt.java:75)
	at org.dragonet.common.utilities.BinaryStream.getUnsignedVarInt(BinaryStream.java:380)
	at org.dragonet.common.utilities.BinaryStream.getByteArray(BinaryStream.java:362)
	at org.dragonet.common.utilities.BinaryStream.getString(BinaryStream.java:371)
	at org.dragonet.common.utilities.BinaryStream.getSlot(BinaryStream.java:320)
	at org.dragonet.protocol.packets.MobEquipmentPacket.decodePayload(MobEquipmentPacket.java:35)
	at org.dragonet.protocol.PEPacket.decode(PEPacket.java:34)
	at org.dragonet.protocol.Protocol.decodeSingle(Protocol.java:138)
	at org.dragonet.protocol.Protocol.decode(Protocol.java:114)
	at org.dragonet.proxy.network.PEPacketProcessor.onTick(PEPacketProcessor.java:95)
	at org.dragonet.proxy.network.UpstreamSession.onTick(UpstreamSession.java:486)
	at org.dragonet.proxy.network.SessionRegister.onTick(SessionRegister.java:38)
	at org.dragonet.proxy.DragonProxy.onTick(DragonProxy.java:235)
	at org.dragonet.proxy.TickerThread.run(TickerThread.java:32)
`

**Comment by HoverEpic** [comment]
Fixed in f48b25389ab5e8157b5dec0e5e720c9ce552b9fe

For the second error dupplicate #185, so closing this issue.
